src/classes/com/sun/tdk/jcov/runtime/Collect.java

```diff
@@ -51,8 +51,9 @@ public class Collect {
     private static JCovSaver[] savers = new JCovSaver[MAX_SAVERS];
     private static int nextSaver = 0;
     private static Class<SaverDecorator> extension = null;
-    // This constant is replaced in ANT build script (see files se.replace.properties, me.replace.properties and so on)
-    private final static String saverClassnameString = "/*@BUILD_MODIFIED_SAVER_STRING@*/";
+
+    protected static boolean isVMReady = false;
```

Comment:
should it not be volatile?

Reply:
As things stand, none of the Collect class is thread safe. So I withdraw my comment with an idea that there could be an enhancement later to make jcov thread safe. Which would be more than just fixing this var.